### ISSUE TITLE
WebRTC screen sharing support

### ISSUE THREAD

**Issue by application-developer-DA** [livekit/livekit-server#75]
Thanks for the great project guys! I've read the whole documentation and it looks amazing.

I've got a question regarding the support of the WebRTC screen sharing though (let me know if this is the wrong place to ask question and if I should prefer using Slack for that). Maybe I missed something but while checking the client SDK and server SDKs I have not found a way to start the WebRTC screen sharing. Is it planned to add a support of it?

At the same time I understand that adding such thing to the client library is [probably] not going to be super portable and might [at the beginning at least] be only supported in the SDKs for the web clients because of the way how the screen sharing works across different platforms (iOS for instance has quite a sophisticated set up that would require implementing a broadcasting extension to be able to get the frames).

**Comment by davidzhao** [comment]
Thanks for the kind words!

We do support screen-sharing in the browser. Today it's just publishing a second video track. You are right it's missing from the docs. We'll add it soon!

```typescript
try {
  const captureStream =
    // @ts-ignore
    (await navigator.mediaDevices.getDisplayMedia({
      video: {
        width: VideoPresets.fhd.resolution.width,
        height: VideoPresets.fhd.resolution.height,
      },
    })) as MediaStream;

  room.localParticipant.publishTrack(captureStream.getTracks()[0], {
    name: "screen",
    videoEncoding: {
      maxBitrate: 3000000,
      maxFramerate: 30,
    },
  });
} catch (err) {
  // TODO: display error
}
```

You can see this in action with the [react example app](https://example.livekit.io)

**Comment by leslie-heinzen** [comment]
@davidzhao Starting a screenshare works well, but it looks like there isn't a way to listen for the mediaStreamTrack's ended event, like when a user clicks the native browser "stop sharing" button. The example react app gets into a weird state if you do this.

You could set up a handler on the underlying mediaStreamTrack, which would fire even with the native browser action, and allow you to detach/unpublish/etc:

```ts
// ...
const screenTrack = new LocalVideoTrack(captureStream.getTracks()[0]);

screenTrack.mediaStreamTrack.onended = () => {
  screenTrack.detach();
  room.localParticipant.unpublishTrack(screenTrack);
};

room.localParticipant.publishTrack(screenTrack, {
// ...
```

You might consider emitting an `ended`/`stopped` event in the js client sdk in a LocalTrack's `mediaStreamTrack.onended` handler so it could be used directly on a LocalTrack instance, unless you don't mind devs modifying the underlying mediaStreamTrack. Personally I don't mind the above, especially since it's pretty specific to screensharing.

This whole project is great, by the way -- keep up the good work!

**Comment by davidzhao** [comment]
@leslie-heinzen, thanks for the report and proposed fix! I think we should handle ended events for all local tracks (since there are other ways for the user to interact with them). I filed an issue for the JS SDK and we'll take a look shortly.

**Comment by davidzhao** [comment]
folks, screen share APIs have been added to the JS SDK: https://github.com/livekit/client-sdk-js/pull/31
